### PR TITLE
fix only item scroll in combobox

### PR DESCRIPTION
**Pull request by Fercas123**
closes https://github.com/jpmorganchase/salt-ds/issues/3618

### CODE REVIEW

File: packages/core/src/option/Option.css [hunk 1]
```diff
@@ -38,22 +38,27 @@
   content: "";
   display: block;
   position: absolute;
-  top: -1px;
-  bottom: -1px;
+  top: -1px; /* HERE */
+  bottom: -1px; /* HERE */
```
Comment (Fercas123) [inline packages/core/src/option/Option.css:42]:
```suggestion
  top: -1px;
  bottom: -1px;
```

**Review by libertymayc — DISMISSED** [review]

File: packages/core/src/option/Option.css [hunk 2]
```diff
@@ -43,17 +43,22 @@
   left: 0;
   width: 100%;
 }
+.saltOption:only-of-type::after {
```
Comment (joshwooding) [inline packages/core/src/option/Option.css:46]: Can you test how this works when only one option is in a group please

**Review by joshwooding — DISMISSED** [review]
Needs some more checks

File: packages/core/stories/combo-box/combo-box.qa.stories.tsx [hunk 3]
```diff
@@ -213,3 +213,26 @@ ClosedExamples.parameters = {
     },
   },
 };
+
+export const OpenWithSingleSelectionExamples: StoryFn<
+  QAContainerProps
+> = () => (
+  <QAContainer cols={3} height={1000}>
```
Comment (Fercas123) [inline packages/core/stories/combo-box/combo-box.qa.stories.tsx:220]:
```suggestion
  <QAContainer cols={3} height={1000} vertical>
```

File: packages/core/stories/combo-box/combo-box.qa.stories.tsx [hunk 4]
```diff
@@ -213,3 +214,28 @@ ClosedExamples.parameters = {
...
+      </ComboBox>
+      <ComboBox multiselect open defaultSelected={["Alaska"]}>
+        <Option value={"Alaska"} />
+      </ComboBox>
+    </StackLayout>
+    <ComboBox open defaultSelected={["Alaska"]} style={{ marginBottom: 200 }}>
+      <OptionGroup label={"A"}>
+        <Option value={"Alaska"} />
+      </OptionGroup>
+      <OptionGroup label={"C"}></OptionGroup>
```
Comment (joshwooding) [inline packages/core/stories/combo-box/combo-box.qa.stories.tsx:234]: Is this group intentionally empty?

Comment (Fercas123) [reply packages/core/stories/combo-box/combo-box.qa.stories.tsx:234]: yes, it was just to make sure the item border was not going to overlap with the group border

**Review by joshwooding — DISMISSED** [review]
The visual regression looks odd to me

File: packages/core/stories/combo-box/combo-box.qa.stories.tsx (hunk 4 above)
Comment (joshwooding) [reply packages/core/stories/combo-box/combo-box.qa.stories.tsx:234]: and I guess adding a option would make the list longer?

Comment (joshwooding) [reply packages/core/stories/combo-box/combo-box.qa.stories.tsx:234]: We should add a comment, it seems something to easily miss.

File: packages/core/src/option/Option.css [hunk 5]
```diff
@@ -32,30 +32,8 @@
...
-}
-
-.saltOption[aria-selected="true"]:first-of-type::before {
-  border-top: unset;
-  top: 0;
-}
-.saltOption[aria-selected="true"]:last-of-type::before {
-  border-bottom: unset;
-  bottom: 0;
+  box-shadow: calc(var(--salt-size-border) * -2) 0 0 0 var(--salt-selectable-background-hover),
```
Comment (joshwooding) [inline packages/core/src/option/Option.css:35]: Needs a changeset

File: packages/core/stories/combo-box/combo-box.qa.stories.tsx [hunk 6]
```diff
@@ -213,3 +214,29 @@ ClosedExamples.parameters = {
     },
   },
 };
+
+export const OpenWithSingleSelectionExamples: StoryFn<
+  QAContainerProps
+> = () => (
+  <QAContainer cols={2} itemPadding={12} width={800} vertical>
+    <StackLayout gap={8.5}>
```
Comment (Fercas123) [inline packages/core/stories/combo-box/combo-box.qa.stories.tsx:222]:
```suggestion
    <StackLayout gap={10}>
```

File: packages/core/stories/dropdown/dropdown.qa.stories.tsx [hunk 7]
```diff
@@ -136,3 +137,29 @@ ClosedExamples.parameters = {
     },
   },
 };
+
+export const OpenWithSingleSelectionExamples: StoryFn<
+  QAContainerProps
+> = () => (
+  <QAContainer cols={2} itemPadding={12} width={800} vertical>
+    <StackLayout gap={8.5}>
```
Comment (Fercas123) [inline packages/core/stories/dropdown/dropdown.qa.stories.tsx:145]:
```suggestion
    <StackLayout gap={10}>
```

**Comment by joshwooding** [comment]
Not super noticeable but this was caught by the visual diff
image
The blue on the right

**Review by joshwooding — APPROVED** [review]

**Review by tomhazledine — APPROVED** [review]
Fixes the issue 👍

**Pull request merged**